src/domain/session/rightpanel/RightPanelViewModel.js

```diff
@@ -0,0 +1,62 @@
+import {ViewModel} from "../../ViewModel.js";
+import {RoomDetailsViewModel} from "./RoomDetailsViewModel.js";
+import {MemberListViewModel} from "./MemberListViewModel.js";
+
+export class RightPanelViewModel extends ViewModel {
+    constructor(options) {
+        super(options);
+        this._room = options.room;
+        this._setupNavigation();
+    }
+
+    get activeViewModel() { return this._activeViewModel; }
+
+    async _getMemberArguments() {
+        const members = await this._room.loadMemberList();
+        const room = this._room;
+        const powerLevelsObservable = await this._room.observePowerLevels();
+        return {members, powerLevelsObservable, mediaRepository: room.mediaRepository};
+    }
+
+    _setupNavigation() {
+        this._hookUpdaterToSegment("details", RoomDetailsViewModel, () => { return {room: this._room}; });
+        this._hookUpdaterToSegment("members", MemberListViewModel, () => this._getMemberArguments());
+    }
+
+    _hookUpdaterToSegment(segment, viewmodel, argCreator) {
+        const observable = this.navigation.observe(segment);
+        const updater = this._setupUpdater(segment, viewmodel, argCreator);
+        this.track(observable.subscribe(() => updater()));
+    }
+
+    _setupUpdater(segment, viewmodel, argCreator) {
+        const updater = async (skipDispose = false) => {
```

Comment:
We have to run the `updater` for both the segments ("memberlist" and "details") once initially because otherwise by the time the `RightPanelViewModel` is created and the segment is observed, the navigation event would have already fired.
Only one of the initial `updater` runs would actually create the `activeViewModel` but we don't want the second initial updater call to dispose of the view model created by an earlier initial updater call.

Reply:
right, makes sense now :+1: